x-pack/plugins/reporting/server/export_types/csv_searchsource_immediate/csv_searchsource_immediate.ts

```diff
@@ -72,9 +72,15 @@ export class CsvSearchSourceImmediateExportType extends ExportType<
     this.logger = this.logger.get('csv-searchsource-export');
   }
 
-  public createJob = async () => {
-    throw new Error(`immediate download has no create job handler!`);
+  // @ts-ignore expected to have a type failure due to deprecated export type
+  public createJob = async (immediateJobParams: any): Promise<JobParamsDownloadCSV> => {
+    return {
+      ...immediateJobParams,
+      title: immediateJobParams.title,
+      objectType: 'immediate-search',
+    };
   };
+
```

Comment:
The code in `x-pack/plugins/reporting/server/export_types` should be untouched in this PR.
```
git checkout main x-pack/plugins/reporting/server/export_types/
```